<launch>

    <!-- MASTER NODE! -->
    <node name="ros_test_node" pkg="camera_imu_calibration" type="ros_test_node" output="screen" clear_params="true" required="true">
        <!-- bag topics -->
        <param name="topic_imu"    type="string" value="/vectornav/IMU" />
        <param name="topic_image"  type="string" value="/pylon_camera_node/image_raw" />

        <!-- bag parameters -->
<!--         <param name = "path_bag" type="string" value="/home/smishr30/Downloads/calibcheckerboard1.bag"/> -->
<!--         <param name = "path_bag" type="string" value="/home/usl/Downloads/cam_imu_calibration/calibcheckerboard2.bag"/> -->
<!--         <param name = "path_bag" type="string" value="/home/smishr30/Downloads/cam_imu_lidar_calib_data/2022-01-28-15-11-20.bag"/> -->
<!--         <param name = "path_bag" type="string" value="/home/smishr30/Downloads/cam_imu_lidar_calib_data/2022-01-28-15-15-06.bag"/> -->
<!--         <param name = "path_bag" type="string" value="/home/smishr30/Downloads/cam_imu_lidar_calib_data/2022-01-28-15-20-21.bag"/> -->
<!--         <param name = "path_bag" type="string" value="/home/smishr30/Downloads/cam_imu_lidar_calib_data/2022-01-28-15-30-47.bag"/> -->
        <param name = "path_bag" type="string" value="/home/smishr30/Downloads/validation/2022-01-30-11-02-08.bag"/>
        <param name="bag_start"   type="double" value="0" />
        <param name="bag_durr"    type="int"    value="-1" />

        <!-- Estimator Parameters -->
        <param name="use_fej" type="bool" value="true" />
        <param name="use_imuavg" type="bool" value="true" />
        <param name="use_rk4int" type="bool" value="true" />
        <param name="calib_lidar_timeoffset" type="bool" value="true" />
        <param name="calib_extrinsics"   type="bool"   value="false" />
        <param name="max_clones" type="int" value="2" />
        <param name="updater_do_chi2_check" type="bool" value="true"/>

        <!-- Filter Init Parameters -->
        <param name="init_window_time" type="double" value="0.5"/> <!-- in second -->
        <param name="init_imu_thresh" type="double" value="0.25"/> <!-- jerk threshold -->

        <!-- Noise Parameters -->
<!--         <param name="accelerometer_noise_density"  type="double"   value="0.01051" /> -->
<!--         <param name="accelerometer_random_walk"    type="double"   value="0.00032" /> -->
<!--         <param name="gyroscope_noise_density"      type="double"   value="0.00062" /> -->
<!--         <param name="gyroscope_random_walk"        type="double"   value="5e-05" /> -->
        <param name="accelerometer_noise_density"  type="double"   value="0.0015825749257056074" />
        <param name="accelerometer_random_walk"    type="double"   value="4.1557212532459854e-05" />
        <param name="gyroscope_noise_density"      type="double"   value="8.97991030686202e-05" />
        <param name="gyroscope_random_walk"        type="double"   value="7.253101866424753e-07" />

        <!-- Updater Parameters -->
        <!-- Chi2 Parameters -->
        <param name="updater_chi2_multiplier" type="int" value="1" />
        <!-- Update Noise (sigma) Params -->
        <param name="updater_translation_noise" type="double" value="0.05" />
        <param name="updater_rotation_noise" type="double" value="0.1" />

        <!-- State Initial Noise (sigma) Params -->
        <param name="state_init_x_noise" type="double" value="0.5" />
        <param name="state_init_y_noise" type="double" value="0.5" />
        <param name="state_init_z_noise" type="double" value="0.5" />
        <param name="state_init_rx_noise" type="double" value="0.5" />
        <param name="state_init_ry_noise" type="double" value="0.5" />
        <param name="state_init_rz_noise" type="double" value="0.5" />

        <param name="state_init_timeoffset_noise" type="double" value="0.01" />

        <!-- Gravity -->
        <rosparam param="gravity">[0.0, 0.0, 9.81]</rosparam>

        <!-- initialization of lidar imu time offset -->
        <param name="calib_cameraimu_dt" type="double" value="0.0" />

        <!-- initial calibration file -->
        <param name="init_calibration_result_filename" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/I_T_C.txt"/>
<!--         <param name="init_calibration_result_filename" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/I_T_C_kalibr.txt"/> -->
<!--         <param name="init_calibration_result_filename" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/I_T_C_cincalib.txt"/> -->

        <!-- Camera Tracking Parameters -->
        <param name="checkerboard_dx" type="double" value="0.0715"/>
        <param name="checkerboard_dy" type="double" value="0.0715"/>
        <param name="checkerboard_rows" type="int" value="6"/>
        <param name="checkerboard_cols" type="int" value="7"/>
        <param name="camera_calibration_file_path" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/pylon_calib.yaml"/>

        <!-- output filenames -->
        <param name="camimu_trajectory_filename" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/camimucalib_trajectory.csv"/>
        <param name="camimu_bias_filename" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/camimucalib_bias.csv"/>
        <param name="camimu_velocity_filename" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/camimucalib_velocity.csv"/>
        <param name="camimu_calib_extrinsic_filename" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/camimu_calib_extrinsic.csv"/>
        <param name="camimu_calib_dt_filename" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/camimu_calib_dt.csv"/>
        <param name="camerapose_trajectory_filename" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/camerapose_trajectory.csv"/>
        <param name="reprojection_error_filename" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/reprojection_error_validation.csv"/>

        <!-- output -->
        <param name="calibration_result_filename" type="string" value="/home/smishr30/catkin_ws/src/camera_imu_calibration/data/I_T_C.txt"/>

    </node>

</launch>
